cmd/state-offline-installer/zip.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"archive/zip"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/unarchiver"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func unpackArtifact(ua unarchiver.Unarchiver, tarballPath string, targetDir string) error {
+	f, i, err := ua.PrepareUnpacking(tarballPath, targetDir)
+	if err != nil {
+		return errs.Wrap(err, "Prepare for unpacking failed")
+	}
+	defer f.Close()
+	return ua.Unarchive(f, i, targetDir)
+}
+
+// Unzip unzips the file zippath and puts it in destination
+func unzip(zippath string, destination string) (err error) {
+	r, err := zip.OpenReader(zippath)
```

Comment:
Are you unable to use the unarchiver? 

https://github.com/ActiveState/cli/blob/cbaa27607e1fc2c8063e11d9bfe10caa146ec0d3/internal/unarchiver/zip.go#L26